include/igl/blue_noise.cpp

```diff
@@ -209,7 +210,9 @@ namespace igl
     }
         //printf("  --------\n");
     // randomize order: this might be a little paranoid...
-    std::random_shuffle(std::begin(N), std::end(N));
+    std::random_device rd;
```

Comment:
Don't initialize with random device, this makes the function nondeterministic. Either use the default seed, or pass the seed as a && argument to the function.

Reply:
Isn't it the point of shuffling to make it nondeterministic?

Reply:
No. This shuffle seems to be here to avoid biasing the sampling due to cell ordering, and it might not even be strictly necessary at that. A random function shouldn't be nondeterministic, those are different concepts.